src/GOODFFrame.cpp

```diff
@@ -675,12 +715,39 @@ void GOODFFrame::OnWriteODF(wxCommandEvent& WXUNUSED(event)) {
 	} else {
 		odfFile->Create(fullFileName);
 	}
+
+	bool isWrittenCorrectly = false;
+	bool isEncodingWorking = false;
+	wxString tail = wxT("!");
 	m_organ->writeOrgan(odfFile);
+	if (IsEncodingOk(odfFile, wxCSConv("ISO-8859-1"))) {
+		isEncodingWorking = true;
+		isWrittenCorrectly = odfFile->Write(wxTextFileType_Dos, wxCSConv("ISO-8859-1"));
+	} else if (IsEncodingOk(odfFile, wxCSConv("UTF-8"))) {
+		isEncodingWorking = true;
+		isWrittenCorrectly = WriteUTF8(odfFile);
+		tail = wxT(" as UTF-8") + tail;
+	}
 
-	odfFile->Write(wxTextFileType_Dos, wxCSConv("ISO-8859-1"));
-	if (!m_organHasBeenSaved) {
-		wxMessageDialog msg(this, wxT("ODF file ") + m_organPanel->getOdfName() + wxT(".organ has been written!"), wxT("ODF file written"), wxOK|wxCENTRE);
-		msg.ShowModal();
+	if (isWrittenCorrectly) {
+		if (!m_organHasBeenSaved) {
+			wxMessageDialog msg(this, wxT("ODF file ") + m_organPanel->getOdfName() + wxT(".organ has been written") + tail, wxT("ODF file written"), wxOK|wxCENTRE);
+			msg.ShowModal();
+		}
+		m_organHasBeenSaved = true;
```

Comment:
I suggest to remove lines 754 to 757 as they are (correctly) moved to 737-740 and only should be executed if the file really was written.